provider/provider.go

```diff
@@ -580,3 +582,30 @@ func getIgnore(dockerIgnorePath string) ([]string, error) {
 	}
 	return ignorePatterns, nil
 }
+
+// setConfiguration takes in the stack config settings and additionally reads in any environment variables
+func setConfiguration(configVars map[string]string) map[string]string {
+	envConfig := make(map[string]string)
+	for key, val := range configVars {
+		envConfig[strings.TrimPrefix(key, "docker:config:")] = val
+	}
+	// get env vars, if any.
+	// Because historically we give precedence to env vars over config vars, we overwrite here.
```

Comment:
Are you sure this is the case? My memory is that it's not the case, in fact it's the inverse. Environment variables are only consulted if the value has not been specified explicitly. `_, ok := envConfig["host"]; !ok`

Reply:
ah, my bad, I misremembered from pulumi-aws :/